pyspark_datasources/restapi.py

```diff
@@ -0,0 +1,402 @@
+import contextlib
+import json
+from ast import literal_eval
+from typing import Any, Generator, Hashable
+from urllib.parse import parse_qs, urlparse, urlunparse
+
+import httpx
+import pyspark.sql.types as T
+import validators
+from authlib.integrations.httpx_client import OAuth2Client
+from authlib.oauth2.rfc7523 import ClientSecretJWT
+from pyspark.sql import Row
+from pyspark.sql.datasource import DataSource, DataSourceReader
+
+
+class RestapiDataSource(DataSource):
```

Comment:
Shall we add some docstring and examples for this data source?

Reply:
Yes, good idea. I added the examples from the tests to the docstring.